### DISCUSSION TITLE
How we get different values of FPS and AP, while on testing we get single value of FPS and AP??

### DISCUSSION THREAD

**Discussion by irfan-gif** [ultralytics/yolov5#2994] [General]
![114313216-f0a5e100-9af5-11eb-8445-c682b60da2e3](https://user-images.githubusercontent.com/54844651/116727632-40ddd800-a9fe-11eb-8c01-3caa34ca42b4.png)

Can some explain me??

**Comment by glenn-jocher** [comment]
@irfan-gif YOLOv5 models can run inference at image sizes from 0 up to the size they were trained on (1280 for the P6 models shown above). Running inference at reduced size will be faster but less accurate. The plot above shows you evaluations of each P6 model across a range of image sizes.

This figure is reproducible using the commands shown in the Figure Notes below the figure:

  * GPU Speed measures end-to-end time per image averaged over 5000 COCO val2017 images using a V100 GPU with batch size 32, and includes image preprocessing, PyTorch FP16 inference, postprocessing and NMS. 
  * EfficientDet data from [google/automl](https://github.com/google/automl) at batch size 8.
  * **Reproduce** by `python test.py --task study --data coco.yaml --iou 0.7 --weights yolov5s6.pt yolov5m6.pt yolov5l6.pt yolov5x6.pt`

**Reply by DailiZhang2010** [reply]
Thanks for the awesome work and for this answer. I had the same question.  :-)
